### PR TITLE
Added a way to hide the tag of admins without removing other features

### PR DESCRIPTION
**Pull request by Sarrus1**
My admins asked me to implement a command to make them anonymous on the server.
The `!toggletag` commands works fine but it removes the tag entirely.
This new `!anonymous` command disables tags for admins and people who had a their steamid referenced in the config, while keeping other tags like Ranks calculated with rankme etc.

**COMPILES** ✅

**TESTED** ✅

### CODE REVIEW

**Comment by fejmek** [comment]
When the tag is hidden with sm_anonymous and hidden admin join the server, such an error will appear:
```
L 04/01/2021 - 14:50:06: [SM] Exception reported: Client index 2 is not valid
L 04/01/2021 - 14:50:06: [SM] Blaming: hextags.smx
L 04/01/2021 - 14:50:06: [SM] Call stack trace:
L 04/01/2021 - 14:50:06: [SM] [0] CS_GetClientClanTag
L 04/01/2021 - 14:50:06: [SM] [1] Line 542, /Users/charleszablit/Documents/Perso/Dev/Sourcemod/HexTags/addons/sourcemod/scripting/hextags.sp::OnClientCookiesCached
```

**Comment by Sarrus1** [comment]
I'll try to take a look tomorrow. In the meantime, do you mine cropping the path of the file so that it doesn't include the username, like : `Line 542, HexTags/addons/sourcemod/scripting/hextags.sp::OnClientCookiesCached` so it doesn't contain personal informations?

**Comment by Sarrus1** [comment]
@fejmek Try with the latest build from my branch [here](https://github.com/Sarrus1/HexTags/blob/master/addons/sourcemod/plugins/hextags.smx). I don't think @Hexer10 is going to merge this.

**Comment by fejmek** [comment]
@Sarrus1 error resolved :D

File: addons/sourcemod/scripting/hextags.sp [hunk 1]
```diff
@@ -137,16 +140,21 @@ public void OnPluginStart()
 	//Reg Cmds
 	RegAdminCmd("sm_reloadtags", Cmd_ReloadTags, ADMFLAG_GENERIC, "Reload HexTags plugin config");
 	RegAdminCmd("sm_toggletags", Cmd_ToggleTags, ADMFLAG_GENERIC, "Toggle the visibility of your tags");
+	RegAdminCmd("sm_anonymous", Cmd_Anonymous, ADMFLAG_GENERIC|ADMFLAG_CUSTOM6, "This allows admins to hide their tags");
 	RegConsoleCmd("sm_tagslist", Cmd_TagsList, "Select your tag!");
 	RegConsoleCmd("sm_getteam", Cmd_GetTeam, "Get current team name");
 
+	g_hAnonymousCookie = RegClientCookie("AnonymousCookie", "Plugin that defines wether or not an admin is anonymous.", CookieAccess_Protected);
```
Comment (Hexer10) [inline addons/sourcemod/scripting/hextags.sp:147]: I would stick with the `HexTags_` prefix for the cookies

File: addons/sourcemod/scripting/hextags.sp [hunk 2]
```diff
@@ -137,16 +140,21 @@ public void OnPluginStart()
 	//Reg Cmds
 	RegAdminCmd("sm_reloadtags", Cmd_ReloadTags, ADMFLAG_GENERIC, "Reload HexTags plugin config");
 	RegAdminCmd("sm_toggletags", Cmd_ToggleTags, ADMFLAG_GENERIC, "Toggle the visibility of your tags");
+	RegAdminCmd("sm_anonymous", Cmd_Anonymous, ADMFLAG_GENERIC|ADMFLAG_CUSTOM6, "This allows admins to hide their tags");
```
Comment (Hexer10) [inline addons/sourcemod/scripting/hextags.sp:143]: I think it's better to just leave the `ADMFLAG_GENERIC` flag and if users want to change the permission they could set it in the admin override

File: addons/sourcemod/scripting/hextags.sp [hunk 3]
```diff
@@ -303,6 +312,49 @@ public void warden_OnWardenRemoved(int client)
...
+			LoadTags(client);
+			char sTag[32];
+			CS_GetClientClanTag(client, sTag, sizeof(sTag));
+			ReplyToCommand(client, "[SM] You are now anonymous. Your tag is %s", selectedTags[client].ScoreTag);
+			if (StrEqual(sTag, selectedTags[client].ScoreTag))
+			{
+				return Plugin_Handled;
+			}
+			CS_SetClientClanTag(client, selectedTags[client].ScoreTag);
+		}
```
Comment (Hexer10) [inline addons/sourcemod/scripting/hextags.sp:353]: The code in the two if branches is pretty much the same except for the Chat  messages so you could just to smth like:
```c
cookieValue = !cookieValue;
g_hAnonymous[client] = cookieValue == 1;
IntToString(cookieValue, sCookieValue, sizeof(sCookieValue));
SetClientCookie(client, hVibilityAdminsCookie, sCookieValue);
LoadTags(client);
char sTag[32];
CS_GetClientClanTag(client, sTag, sizeof(sTag));
if (!cookieValue)
{
	ReplyToCommand(client, "[SM] You are no longer anonymous. Your tag is %s", selectedTags[client].ScoreTag);
} 
else 
{
	ReplyToCommand(client, "[SM] You are now anonymous. Your tag is %s", selectedTags[client].ScoreTag);
}		
if (StrEqual(sTag, selectedTags[client].ScoreTag))
{
	return Plugin_Handled;
}
CS_SetClientClanTag(client, selectedTags[client].ScoreTag);
```

**Review by Hexer10 — CHANGES_REQUESTED** [review]
Sorry for the time it took to review this. Right now I'm having school exams, and I started focusing on other projects, so I had little to no time to work on this.

However your PR looks file, there are just some small quirks to fix and I'll merge it

**Pull request merged**